src/functionalTest/groovy/com/autonomousapps/jvm/ComponentWithMultipleArtifactsSpec.groovy

```diff
@@ -0,0 +1,40 @@
+package com.autonomousapps.jvm
+
+import com.autonomousapps.jvm.projects.JarTransformingProject
+import com.autonomousapps.jvm.projects.MultipleJarsProject
+
+import static com.autonomousapps.utils.Runner.build
+import static com.google.common.truth.Truth.assertThat
+
+final class ComponentWithMultipleArtifactsSpec extends AbstractJvmSpec {
+
+  def "one component can have multiple Jars (#gradleVersion)"() {
```

Comment:
I know this is an obvious question, but this test would fail without the changes you made? It's not clear to me which change would cause the test to switch from failing to passing.

Reply:
Yes it fails (both new tests fail).

The important changes are these two `compareTo` implementations:

https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/688/files#diff-a7da78989269bbf240e2f6adc5f5e1a6ac7c18bcf0ba4b3bc1b2fddad041defeR197

https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/688/files#diff-8f376c467c8ba4a5dcbe0c0dadc0d561d71916cd68a5afb0781f1d63c0421776R17

There are two places where a list is converted into an OrderedSet. Without this changes, the OrderedSets only have one entry per _coordinates_. If there are more than one file for one _coordinates_, entries will be dropped with the old _compareTo_ implementations.

Here are the places where the sets are created:

https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/blob/03a4ad7e6655beaecb75b1dabd85c4b87af08d8d/src/main/kotlin/com/autonomousapps/tasks/ArtifactsReportTask.kt#L88

https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/blob/33c25ca68b248ae13fab454f05c580115f7dfbb8/src/main/kotlin/com/autonomousapps/internal/JarExploder.kt#L31


**Note:** The change in `Dependency.kt` does not have any effect, because the code seems to be unused. But I would do it, because otherwise it is inconsistent. Another option would be to remove the `Dependency.file(s)` field completely if it is not needed:
 https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/688/files#diff-9e4cd8d8f8a7cf963f2c64cc6f1e5c071bb9f6c2378d2e59bf874f9bf9b95ba8R13